AsyncDisplayKit/ASPagerFlowLayout.m

```diff
@@ -75,19 +56,49 @@ - (CGPoint)_targetContentOffsetForItemAtIndexPath:(NSIndexPath *)indexPath propo
     return proposedContentOffset;
   }
   
-  UICollectionViewLayoutAttributes *attributes = [self layoutAttributesForItemAtIndexPath:_currentIndexPath];
+  UICollectionViewLayoutAttributes *attributes = [self layoutAttributesForItemAtIndexPath:indexPath];
   if (attributes == nil) {
     return proposedContentOffset;
   }
-  
+
   CGFloat xOffset = (CGRectGetWidth(self.collectionView.bounds) - CGRectGetWidth(attributes.frame)) / 2.0;
   return CGPointMake(attributes.frame.origin.x - xOffset, proposedContentOffset.y);
 }
 
 - (BOOL)_dataSourceIsEmpty
 {
-    return ([self.collectionView numberOfSections] == 0 ||
-            [self.collectionView numberOfItemsInSection:0] == 0);
+  return ([self.collectionView numberOfSections] == 0 ||
+          [self.collectionView numberOfItemsInSection:0] == 0);
+}
+
+- (void)_updateCurrentNode
+{
+  // Never change node during an animated bounds change (rotation)
+  // NOTE! Listening for -prepareForAnimatedBoundsChange and -finalizeAnimatedBoundsChange
+  // isn't sufficient here! It's broken!
+  NSArray *animKeys = self.collectionView.layer.animationKeys;
+  for (NSString *key in animKeys) {
+    if ([key hasPrefix:@"bounds"]) {
+      return;
+    }
+  }
+  
+  CGRect bounds = self.collectionView.bounds;
+  CGRect rect = CGRectMake(CGRectGetMidX(bounds), CGRectGetMidY(bounds), 1, 1);
```

Comment:
Probably doesn't matter, but should this possibly be `CGRectMake(CGRectGetMidX(bounds) - 0.5, CGRectGetMidY(bounds) - 0.5, 1, 1);`